src/openloco/input/mouse_input.cpp

```diff
@@ -211,6 +211,36 @@ namespace openloco::input
         return (*_toolWindowType == type && _toolWindowNumber == number);
     }
 
+    // 0x004CE367
+    // tool (al)
+    // widgetIndex (dx)
+    // w (esi)
+    // TODO: Maybe create a an enum similar to TOOL_IDX in OpenRCT2 for tool (instead of uint8_t)
+    bool tool_set(ui::window* w, int16_t widgetIndex, uint8_t tool)
+    {
+        if (input::has_flag(input::input_flags::tool_active))
+        {
+            if (w->type == _toolWindowType && w->number == _toolWindowNumber
+                && widgetIndex == _toolWidgetIndex)
+            {
+                cancel_tool();
+                return true;
+            }
+            else
+            {
+                cancel_tool();
+            }
+        }
```

Comment:
This appears to be a variant of `cancel_tool`. Perhaps this can be refactored to be a third signature for it (including the widget index the current ones are missing).

Reply:
It seems it's not worth to extract it due to returning the bool value ('return true' part).